### ISSUE TITLE
Exposed token in `http://localhost:<port>/neutralino.js` route

### ISSUE THREAD

**Issue by raffaelj** [neutralinojs/neutralinojs#351]
I searched for a security policy and `SECURITY.md` pointed me to file a new issue. I would have sent a private mail instead...

I fiddled with the window mode and an enabled inspector to watch the network stream. In parallel, I started Postman to test, what happens, if I call Nutralino routes from a different software.

Calling `http://localhost:5006/filesystem/readFile` without an auth header closed the app with `Segmentation fault (core dumped)` --> good candidate for a different issue to prevent app crashes on random unauthorized api calls.

Next step was to get the `NL_TOKEN`. After calling `http://localhost:5006/neutralino.js` I already had everything I need to get full filesystem access with my current user.

I'm not that deep into exploits, so I'm not sure, how to automate an attack in this way. Access from different apps might not be harmful, since they already have file system rights and I have to trust them before installation. A quick test on a remote machine failed due to Same Origin Policy, but it still feels bad.

I'm aware, that I also would have to run a port scan in production without the hardcoded port.

Test file on a random webhost:

```html
<!DOCTYPE html>
<html><head><meta charset="UTF-8"><title>test</title></head><body>

<!-- exposed token, but only on my own machine -->
<iframe src="http://127.0.0.1:5006/neutralino.js"></iframe>

<!-- failed due to CORS -->
<script>
window.fetch('http://127.0.0.1:5006/neutralino.js').then(data => console.log(data));
</script>
</body></html>
```

**Steps to Reproduce the Problem**

1. Download and extract latest release
2. `cd path/to/app` and run app `./neutralino-linux`
3. Use Postman or curl etc. to send a GET request to `http://localhost:5006/neutralino.js`
4. Extract `NL_TOKEN` from response
5. Send a POST request to `http://localhost:5006/filesystem/readFile`, header `Authorization: Basic xxTokenxx`, raw json body `{"filename":"/any/file/with/read/permission"}`

**Specifications**

* NeutralinoJs Version: 1.6.0
* Platform: OpenSUSE Tumbleweed/KDE Plasma5

**Comment by raffaelj** [comment]
I found a temporary workaround to prevent too much damage by sandboxing `neutralino-linux` using [firejail][1].

My setup isn't fully tested and your setup might differ due to os differences.

Steps to reproduce (on OpenSUSE Tumbleweed):

1. install firejail and add user group

```bash
sudo zypper install firejail
sudo usermod -a -G firejail $USER
# now logout and login again
```

2. Create a profile and name it `neutralino-linux.profile`

```bash
# Copy this file into ~/.config/firejail
# Not fully tested, it's just a quick and dirty mix from `default.profile` and `electron.profile`

# Persistent local customizations
include default.local
# Persistent global definitions
include globals.local

include disable-common.inc
include disable-devel.inc

# Prevents executing ALL binaries in home dir
# include disable-exec.inc

# Copy/paste of disable-exec.inc with disabled noexec ${HOME}
# noexec ${HOME}
noexec ${RUNUSER}
noexec /dev/shm
noexec /tmp
# /var is noexec by default for unprivileged users
# except there is a writable-var option, so just in case:
noexec /var


include disable-interpreters.inc
include disable-passwdmgr.inc
include disable-programs.inc
include disable-xdg.inc

# Might be useful
# apparmor

# caps.keep sys_admin,sys_chroot
caps.drop all

net none

netfilter
nodvd
nogroups
notv
nou2f
novideo
shell none

disable-mnt
private-cache
private-dev
private-tmp

dbus-user none
dbus-system none
```

3. `cd` into the project folder and run `firejail --whitelist=$PWD ./neutralino-linux`
4. Optionally create a shortcut, e. g. `run.sh` and make it executable `chmod +x ./neutralino-linux`

```bash
#!/bin/bash
firejail --whitelist=$PWD ./neutralino-linux
```

Now Neutralino shouldn't be able to write outside of the current working directory. A simple test with `storage.putData()` didn't effect the parent folder anymore, so if any other application takes control over an exposed Neutralino dev project, the damage should stay in the current project folder.

```js
Neutralino.storage.putData({bucket:"../../test",content:{}}, function () {
    console.log('success, even if not able to write');
}, function () {
    console.log('fail, but it seems to always return success');
});
```

[1]: https://github.com/netblue30/firejail

**Comment by shalithasuranga** [comment]
Thank you very much for finding this security problem. As you already noticed, many things have been changed in Neutralinojs. Now we use a WebSocket connection. Also, we've recently added a `one-time` mode for the token, so `NL_TOKEN` is not exposed every time. You can check further details about this feature from [here](https://github.com/neutralinojs/neutralinojs/blob/main/CHANGELOG.md#core-auth)

I am going to close this issue because of the `one-time` feature and some outdated details about Neutralinojs, but feel free to discuss more details. Thanks again for discussing this :)
